### PR TITLE
Rework the correctness validation Claude skill

### PR DESCRIPTION
**Pull request by haok1402**
- invoke setup-benchmark-inputs and launch-with-slurm inside the validate-correctness skills.
- bump the tolerance to 5e-3 since the same code running again has some deviance in the loss where in the largest case 3.23e-3, which if the previous 1e-3 threshold was enforced, wouldn't pass.

### CODE REVIEW

File: .claude/skills/validate-correctness/scripts/compare.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+"""Diff two PithTrain training logs: gate on correctness, report on performance."""
+
+import argparse
+import re
+import sys
+from statistics import median, stdev
+
+CORRECTNESS_METRICS = ["cross-entropy-loss", "load-balance-loss"]
```
[the remark this answers is not in the document]
Comment (haok1402) [reply .claude/skills/validate-correctness/scripts/compare.py:8]: yeah ; we need to remove that ; using loss directly ; there is noise in gradient norm that's hard to set a tolerance.

File: .claude/skills/validate-correctness/scripts/compare.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    if not base_steps:
+        print("FAIL: No training steps found in base log.")
+        sys.exit(1)
+    if not feature_steps:
+        print("FAIL: No training steps found in feature log.")
+        sys.exit(1)
+    if len(base_steps) != len(feature_steps):
+        n = min(len(base_steps), len(feature_steps))
+        print(f"WARNING: Step count mismatch — base has {len(base_steps)}, feature has {len(feature_steps)}. Comparing first {n} steps.\n")
```
[the remark this answers is not in the document]
Comment (haok1402) [reply .claude/skills/validate-correctness/scripts/compare.py]: they will have the same step count. we enforce 25 steps.

File: .claude/skills/validate-correctness/scripts/validate.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+training = cfg.training
+training.model = Path(specs["config"])
+training.optimizer = "Adam"
+training.scheduler = "Constant"
+training.max_lr = 1e-6
+training.min_lr = 1e-6
+training.warmup_steps = 0
+training.max_steps = parsed.max_steps
+training.micro_batch_size = 1
+training.global_batch_size = 1024
```
[the remark this answers is not in the document]
Comment (haok1402) [reply .claude/skills/validate-correctness/scripts/validate.py:52]: okay ; in most situations, where EP=8 1024 can be perfectly divided. don't need to worry about it now.

**Review by MasterJH5574 — APPROVED** [review]
